### PR TITLE
Add IrRegex library to vendor/purs/runtime

### PR DESCRIPTION
**Pull request by postsolar**
This is a slightly modified version of [IrRegex](https://github.com/ashinn/irregex) library by Alex Shinn. The license is "BSD-like" per the source code.

The only changes are:
- Modified the `error` definition in `irregex.chezscheme.sls` and `irregex-utils.chezscheme.sls` (they are duplicates) so that they don't raise non-continuables
- Removed unnecessary files
- As usual, added the `purs runtime ...` prefix to the library names

This library powers the ongoing FFI implementation for `purescript-strings`.

### CODE REVIEW

File: vendor/purs/runtime/irregex/irregex.scm [hunk 1]
```diff
@@ -0,0 +1,3984 @@
...
+;; "No, it's not quite so bad as that.  It is the unofficial
+;; force, -- the Baker Street irregulars."
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;; Notes
+;;
+;; This code should not require any porting - it should work out of
+;; the box in any R[457]RS Scheme implementation.  Slight modifications
+;; are needed for R6RS (a separate R6RS-compatible version is included
+;; in the distribution as irregex-r6rs.scm).
```
Comment (JordanMartinez) [inline vendor/purs/runtime/irregex/irregex.scm:23]: Do we need to include `irregex-r6rs.scm` as well?

Comment (postsolar) [reply vendor/purs/runtime/irregex/irregex.scm:23]: > Do we need to include `irregex-r6rs.scm` as well?

According to [Community Scheme Wiki](http://community.schemewiki.org/?Scheme-FAQ-Standards) Chez complies with R5RS. I'm not versed in Scheme standards though.

Comment (postsolar) [reply vendor/purs/runtime/irregex/irregex.scm:23]: Latest CSUG [claims support for R5RS](https://cisco.github.io/ChezScheme/csug9.6/libraries.html#./libraries:h1).

Comment (f-f) [reply vendor/purs/runtime/irregex/irregex.scm:23]: That link talks about R6RS though? 🤔

Comment (postsolar) [reply vendor/purs/runtime/irregex/irregex.scm:23]: There's `(rnrs r5rs (6))` in the list of built-in libraries.

Comment (f-f) [reply vendor/purs/runtime/irregex/irregex.scm:23]: I guess we could try it out this way and see if it works fine

**Review by anttih — APPROVED** [review]

**Review by f-f — APPROVED** [review]

**Pull request merged**